File: leetcode/2021/february/practice/java/1480-RunningSumof1dArray.java
import java.util.Arrays;

class Solution {
    public static int[] runningSum(int[] nums) { 
        for(int i=1;i<nums.length;i++){
            nums[i]=nums[i-1]+nums[i];
        }
        return nums;
    }

    public static void main(String[] args) {
        int[] arr = {1,2,3,4};
        System.out.println(Arrays.toString(runningSum(arr)));;
    }
}

File: leetcode/2021/february/practice/cpp/1480-RunningSumof1dArray.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        for (int i(1); i < nums.size(); i++)
            nums[i] = nums[i - 1] + nums[i];
        return nums;
    }
};

int main() {
    Solution solution;
    vector<int> nums{ 1,1,1,1,1 };
    for (int num : solution.runningSum(nums))
        cout << num << " ";
    return 0;
}